### PR TITLE
[New Rule] Default Cobalt Strike Team Server Certificate 

### LINKED ISSUE #357: [New Rule] Default Cobalt Strike Team Server Certificate
## Description
Cobalt Strike is software for Adversary Simulations and Red Team Operations which are security assessments that replicate the tactics and techniques of an advanced adversary in a network.

While Cobalt Strike is intended to be used for penetration tests and IR training, it is frequently used by actual threat actors (TA) such as APT19, APT29, APT32, APT41, FIN6, DarkHydrus, CopyKittens, Cobalt Group, Leviathan, and many other unnamed criminal TAs.

## Required Info

- **Eventing Sources:**
Filebeat (Zeek module)
Packetbeat

- **Target Operating Systems:**
NA

- **Platforms**
NA

- **Target ECS Version:** 1.6.0
- **New fields required in ECS for this?** NA
- **Related issues or PRs** 

## Optional Info
- **References:**
https://attack.mitre.org/software/S0154/

## Example Data

image
image
```
{
  "_index": "packetbeat-7.9.2-2020.10.02-000001",
  "_type": "_doc",
  "_id": "N51f_nQBjIdSkzxcKonH",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-10-05T16:38:29.687Z",
    "ecs": {
      "version": "1.5.0"
    },
    "client": {
      "ip": "[redacted]",
      "port": 44434
    },
    "tls": {
      "server": {
        "hash": {
          "md5": "950098276A495286EB2A2556FBAB6D83",
          "sha1": "6ECE5ECE4192683D2D84E25B0BA7E04F9CB7EB7C",
          "sha256": "87F2085C32B6A2CC709B365F55873E207A9CAA10BFFECF2FD16D3CF9D94D390C"
        },
        "x509": {
          "signature_algorithm": "SHA256-RSA",
          "version": 3,
          "issuer": {
            "distinguished_name": "OU=,O=,L=,ST=,C="
          },
          "not_after": "2025-05-17T18:26:24.000Z",
          "public_key_size": 2048,
          "public_key_algorithm": "RSA",
          "serial_number": "146473198",
          "subject": {
            "distinguished_name": "OU=,O=,L=,ST=,C="
          },
          "not_before": "2015-05-20T18:26:24.000Z",
          "version_number": 3
        },
        "subject": "OU=,O=,L=,ST=,C=",
        "issuer": "OU=,O=,L=,ST=,C=",
        "not_before": "2015-05-20T18:26:24.000Z",
        "not_after": "2025-05-17T18:26:24.000Z"
      },
      "detailed": {
        "server_hello": {
          "version": "3.3",
          "session_id": "5f7b4c056aebaacb03ac1ee1af1732b292df759ddeb310d65b17eea36e1a849d",
          "selected_compression_method": "NULL",
          "extensions": {
            "_unparsed_": [
              "renegotiation_info"
            ]
          }
        },
        "server_certificate": {
          "signature_algorithm": "SHA256-RSA",
          "version_number": 3,
          "public_key_size": 2048,
          "version": 3,
          "serial_number": "146473198",
          "public_key_algorithm": "RSA",
          "not_before": "2015-05-20T18:26:24.000Z",
          "not_after": "2025-05-17T18:26:24.000Z",
          "issuer": {
            "distinguished_name": "OU=,O=,L=,ST=,C="
          },
          "subject": {
            "distinguished_name": "OU=,O=,L=,ST=,C="
          }
        },
        "client_certificate_requested": false,
        "version": "TLS 1.2",
        "client_hello": {
          "extensions": {
            "_unparsed_": [
              "15"
            ],
            "ec_points_formats": [
              "uncompressed",
              "ansiX962_compressed_prime",
              "ansiX962_compressed_char2"
            ],
            "supported_groups": [
              "secp256r1",
              "secp521r1",
              "brainpoolP512r1",
              "brainpoolP384r1",
              "secp384r1",
              "brainpoolP256r1",
              "secp256k1",
              "sect571r1",
              "sect571k1",
              "sect409k1",
              "sect409r1",
              "sect283k1",
              "sect283r1"
            ],
            "session_ticket": "",
            "signature_algorithms": [
              "rsa_pkcs1_sha512",
              "(unknown:0x0602)",
              "ecdsa_secp521r1_sha512",
              "rsa_pkcs1_sha384",
              "(unknown:0x0502)",
              "ecdsa_secp384r1_sha384",
              "rsa_pkcs1_sha256",
              "(unknown:0x0402)",
              "ecdsa_secp256r1_sha256",
              "(unknown:0x0301)",
              "(unknown:0x0302)",
              "(unknown:0x0303)",
              "rsa_pkcs1_sha1",
              "(unknown:0x0202)",
              "ecdsa_sha1"
            ]
          },
          "version": "3.3",
          "supported_compression_methods": [
            "NULL"
          ]
        }
      },
      "resumed": false,
      "version": "1.2",
      "version_protocol": "tls",
      "cipher": "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
      "established": true,
      "client": {
        "ja3": "9e3c05dbcd9b65d4a4345763305b3b90",
        "supported_ciphers": [
          "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
          "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
          "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
          "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
          "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
          "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
          "TLS_DHE_DSS_WITH_AES_256_GCM_SHA384",
          "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384",
          "TLS_DHE_RSA_WITH_AES_256_CBC_SHA256",
          "TLS_DHE_DSS_WITH_AES_256_CBC_SHA256",
          "TLS_DHE_RSA_WITH_AES_256_CBC_SHA",
          "TLS_DHE_DSS_WITH_AES_256_CBC_SHA",
          "TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA",
          "TLS_DHE_DSS_WITH_CAMELLIA_256_CBC_SHA",
          "TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384",
          "TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384",
          "TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384",
          "TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384",
          "TLS_ECDH_RSA_WITH_AES_256_CBC_SHA",
          "TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA",
          "TLS_RSA_WITH_AES_256_GCM_SHA384",
          "TLS_RSA_WITH_AES_256_CBC_SHA256",
          "TLS_RSA_WITH_AES_256_CBC_SHA",
          "TLS_RSA_WITH_CAMELLIA_256_CBC_SHA",
          "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
          "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
          "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
          "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
          "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
          "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
          "TLS_DHE_DSS_WITH_AES_128_GCM_SHA256",
          "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
          "TLS_DHE_RSA_WITH_AES_128_CBC_SHA256",
          "TLS_DHE_DSS_WITH_AES_128_CBC_SHA256",
          "TLS_DHE_RSA_WITH_AES_128_CBC_SHA",
          "TLS_DHE_DSS_WITH_AES_128_CBC_SHA",
          "TLS_DHE_RSA_WITH_SEED_CBC_SHA",
          "TLS_DHE_DSS_WITH_SEED_CBC_SHA",
          "TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA",
          "TLS_DHE_DSS_WITH_CAMELLIA_128_CBC_SHA",
          "TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256",
          "TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256",
          "TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256",
          "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256",
          "TLS_ECDH_RSA_WITH_AES_128_CBC_SHA",
          "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA",
          "TLS_RSA_WITH_AES_128_GCM_SHA256",
          "TLS_RSA_WITH_AES_128_CBC_SHA256",
          "TLS_RSA_WITH_AES_128_CBC_SHA",
          "TLS_RSA_WITH_SEED_CBC_SHA",
          "TLS_RSA_WITH_CAMELLIA_128_CBC_SHA",
          "TLS_ECDHE_RSA_WITH_RC4_128_SHA",
          "TLS_ECDHE_ECDSA_WITH_RC4_128_SHA",
          "TLS_ECDH_RSA_WITH_RC4_128_SHA",
          "TLS_ECDH_ECDSA_WITH_RC4_128_SHA",
          "TLS_RSA_WITH_RC4_128_SHA",
          "TLS_RSA_WITH_RC4_128_MD5",
          "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_ECDHE_ECDSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA",
          "TLS_ECDH_RSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_ECDH_ECDSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
          "TLS_EMPTY_RENEGOTIATION_INFO_SCSV"
        ]
      }
    },
    "agent": {
      "ephemeral_id": "e142814a-984f-43e7-ab6f-a5622c65b173",
      "id": "ab255210-3866-4b07-9579-91f3a78eb996",
      "name": "[redacted]",
      "type": "packetbeat",
      "version": "7.9.2",
      "hostname": "[redacted]"
    },
    "tags": [
      "test-tag"
    ],
    "destination": {
      "port": 443,
      "ip": "[redacted]"
    },
    "server": {
      "ip": "[redacted]",
      "port": 443
    },
    "status": "OK",
    "type": "tls",
    "source": {
      "ip": "[redacted]",
      "port": 44434
    },
    "related": {
      "ip": [
        "[redacted]",
        "[redacted]"
      ]
    },
    "event": {
      "kind": "event",
      "dataset": "tls",
      "duration": 442044000,
      "start": "2020-10-05T16:38:29.687Z",
      "end": "2020-10-05T16:38:30.129Z",
      "category": [
        "network_traffic",
        "network"
      ],
      "type": [
        "connection",
        "protocol"
      ]
    },
    "network": {
      "type": "ipv4",
      "transport": "tcp",
      "protocol": "tls",
      "community_id": "1:zg4fNVpkgcK8gfkw3wwWQQbuifc="
    },
    "host": {
      "hostname": "[redacted]",
      "architecture": "x86_64",
      "os": {
        "version": "10.15.6",
        "family": "darwin",
        "name": "Mac OS X",
        "kernel": "19.6.0",
        "build": "19G2021",
        "platform": "darwin"
      },
      "id": "7A1239AF-8E8B-5928-A112-24FDC1718CF9",
      "ip": [
        "fe80::aede:48ff:fe00:1122",
        "fe80::1020:ed8:4a98:bd42",
        "192.168.158.141",
        "fe80::d4b7:66ff:fea9:4936",
        "fe80::d4b7:66ff:fea9:4936",
        "fe80::1378:8247:d95c:96eb",
        "fe80::3d9d:4d7e:77f9:9888",
        "fe80::28ea:4c13:25a5:2109",
        "fe80::7f9b:d94c:bebe:f828",
        "fe80::e361:3ca4:f4d7:8660",
        "fe80::b8b5:e6fb:9e66:414d",
        "192.168.33.1",
        "fe80::8f57:8a64:c203:6861",
        "fe80::d31f:535d:3d7d:330c"
      ],
      "mac": [
        "ac:de:48:00:11:22",
        "a6:83:e7:ac:3c:98",
        "a4:83:e7:ac:3c:98",
        "82:ae:16:c5:30:01",
        "82:ae:16:c5:30:00",
        "82:ae:16:c5:30:05",
        "82:ae:16:c5:30:04",
        "82:ae:16:c5:30:01",
        "06:83:e7:ac:3c:98",
        "d6:b7:66:a9:49:36",
        "d6:b7:66:a9:49:36",
        "0a:00:27:00:00:00"
      ],
      "name": "[redacted]"
    }
  },
  "fields": {
    "tls.server_certificate.not_before": [
      "2015-05-20T18:26:24.000Z"
    ],
    "tls.server_certificate.not_after": [
      "2025-05-17T18:26:24.000Z"
    ],
    "event.end": [
      "2020-10-05T16:38:30.129Z"
    ],
    "@timestamp": [
      "2020-10-05T16:38:29.687Z"
    ],
    "tls.server.not_before": [
      "2015-05-20T18:26:24.000Z"
    ],
    "tls.detailed.server_certificate.not_after": [
      "2025-05-17T18:26:24.000Z"
    ],
    "event.start": [
      "2020-10-05T16:38:29.687Z"
    ],
    "tls.server.not_after": [
      "2025-05-17T18:26:24.000Z"
    ],
    "tls.detailed.server_certificate.not_before": [
      "2015-05-20T18:26:24.000Z"
    ]
  },
  "highlight": {
    "event.category": [
      "@kibana-highlighted-field@network_traffic@/kibana-highlighted-field@",
      "@kibana-highlighted-field@network@/kibana-highlighted-field@"
    ],
    "tls.server.hash.md5": [
      "@kibana-highlighted-field@950098276A495286EB2A2556FBAB6D83@/kibana-highlighted-field@"
    ],
    "tls.server.hash.sha1": [
      "@kibana-highlighted-field@6ECE5ECE4192683D2D84E25B0BA7E04F9CB7EB7C@/kibana-highlighted-field@"
    ],
    "tls.server.hash.sha256": [
      "@kibana-highlighted-field@87F2085C32B6A2CC709B365F55873E207A9CAA10BFFECF2FD16D3CF9D94D390C@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1601915909687
  ]
}
```

### PR DESCRIPTION
**Pull request by peasead**
## Issues
Resolves #357 

## Summary
This rule detects the use of the default Cobalt Strike Team Server TLS certificate. Cobalt Strike is software for
Adversary Simulations and Red Team Operations which are security assessments that replicate the tactics and techniques
of an advanced adversary in a network. 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-tls.html",
+    "https://www.elastic.co/guide/en/beats/filebeat/7.9/filebeat-module-suricata.html",
+    "https://www.elastic.co/guide/en/beats/filebeat/7.9/filebeat-module-zeek.html",
+]
+risk_score = 100
+rule_id = "e7075e8d-a966-458e-a183-85cd331af255"
+severity = "critical"
+tags = [
+    "Command and Control",
+    "Post-execution",
```
Comment (threat-punter) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:33]:
```suggestion
    "Post-Execution",
```

Minor tweak required for test to pass.

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  tls.server.hash.sha256:87F2085C32B6A2CC709B365F55873E207A9CAA10BFFECF2FD16D3CF9D94D390C)
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1071.001"
+name = "Application Layer Protocol: Web Protocols"
+reference = "https://attack.mitre.org/techniques/T1071/001/"
```
Comment (threat-punter) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:52]: Looks like a test is failing due to the ATT&CK metadata 🤔

We might need to check with @brokensound77 on what's expected when referencing sub-techniques in rules.

```
AssertionError: 'Web Protocols' != 'Application Layer Protocol: Web Protocols'
- Web Protocols
+ Application Layer Protocol: Web Protocols
: ATT&CK technique mapping error for rule: e7075e8d-a966-458e-a183-85cd331af255 - Default Cobalt Strike Team Server Certificate ->
expected: Web Protocols for T1071.001
actual: Application Layer Protocol: Web Protocols
```

**Review by threat-punter — COMMENTED** [review]
LGTM pending @brokensound77's comments about the sub-technique metadata

**Review by dstepanic17 — APPROVED** [review]
LGTM, thanks for including the screenshots and data in the issue.

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+[metadata]
+creation_date = "2020/10/05"
+ecs_version = ["1.6.0"]
```
Comment (threat-punter) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:3]:
```suggestion
```

Comment (threat-punter) [reply rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:3]: This is no longer required unless the rule won't work with the latest ECS version.

**Review by threat-punter — APPROVED** [review]

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  tls.server.hash.sha1:6ECE5ECE4192683D2D84E25B0BA7E04F9CB7EB7C or
+  tls.server.hash.sha256:87F2085C32B6A2CC709B365F55873E207A9CAA10BFFECF2FD16D3CF9D94D390C)
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1071.001"
+name = "Application Layer Protocol: Web Protocols"
```
Comment (bm11100) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:50]:
```suggestion
name = "Application Layer Protocol"
```

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+event.category:(network or network_traffic) and (tls.server.hash.md5:950098276A495286EB2A2556FBAB6D83 or
+  tls.server.hash.sha1:6ECE5ECE4192683D2D84E25B0BA7E04F9CB7EB7C or
+  tls.server.hash.sha256:87F2085C32B6A2CC709B365F55873E207A9CAA10BFFECF2FD16D3CF9D94D390C)
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1071.001"
```
Comment (bm11100) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:49]:
```suggestion
id = "T1071"
```

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+name = "Application Layer Protocol: Web Protocols"
+reference = "https://attack.mitre.org/techniques/T1071/001/"
```
Comment (bm11100) [inline rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:51]:
```suggestion
reference = "https://attack.mitre.org/techniques/T1071/"
```

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml (hunk 5 above)
Comment (bm11100) [reply rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:49]: without sub techniques it'll want this to pass

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml (hunk 4 above)
Comment (bm11100) [reply rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:50]: without sub techniques it'll want this to pass

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml (hunk 5 above)
Comment (bm11100) [reply rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:49]: are sub techniques still planned for this release @brokensound77

File: rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml (hunk 6 above)
Comment (bm11100) [reply rules/network/command_and_control_cobalt_strike_default_teamserver_cert.toml:51]: needs committed to pass unit tests

**Pull request merged**